localstack/services/sqs/models.py

```diff
@@ -1154,6 +1157,11 @@ def receive(
                 else:
                     self.inflight.add(message)
 
+        # finally, add all empty groups back to the queue since there are potentially no inflight messages
+        # to put them back
+        for group in empty_groups:
+            self.message_group_queue.put_nowait(group)
```

Comment:
RIght now this is not part of the mutex block since the initial pop from the queue isn't either